src/miner.cpp

```diff
@@ -81,15 +81,13 @@ void BlockAssembler::resetBlock()
     nBlockSigOpsCost = 400;
     fIncludeWitness = false;
 
-    // These counters do not include coinbase tx
-    nBlockTx = 0;
     nFees = 0;
 }
 
 Optional<int64_t> BlockAssembler::m_last_block_num_txs{nullopt};
 Optional<int64_t> BlockAssembler::m_last_block_weight{nullopt};
 
-std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& scriptPubKeyIn)
+std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& scriptPubKeyIn, const std::vector<CTransactionRef>* preselected_txs)
```

Comment:
What do you need `BlockAssembler` for when you have a preselected set of transactions? Can't you construct a `CBlockTemplate` directly?

Reply:
I did initially https://github.com/bitcoin/bitcoin/pull/17653/commits/b0745f8dc517501f612b56a0cee726e5bc4c4c8c but @instagibbs suggested I take this approach instead https://github.com/bitcoin/bitcoin/pull/17653#issuecomment-561225955. Now I'm not sure which is better, duplicating a lot of code from `BlockAssembler` or modifying it to support adding preselected txs.